TransferMethod/Sources/View/Widget/TextWidget.swift

```diff
@@ -107,41 +107,34 @@ class TextWidget: AbstractWidget {
         if let pattern = pattern {
             let currentText = getTextForPatternCharacter(PatternCharacter.lettersAndNumbersPatternCharacter.rawValue,
                                                          inputText)
-            var currentTextIndex = inputText.startIndex
             var finalText = ""
+            var index = CurrentIndex(textIndex: inputText.startIndex, patternIndex: pattern.startIndex)
             var isEscapedCharacter = false
-            var patternIndex = pattern.startIndex
 
             if let currentText = currentText, !currentText.isEmpty {
                 var patternCharactersToBeWritten = ""
 
                 while true {
-                    let patternRange = patternIndex ..< pattern.index(after: patternIndex)
-                    let currentPatternCharacter = [Character](pattern[patternRange])
-                    let currentTextRange = currentTextIndex ..< currentText.index(after: currentTextIndex)
-                    let currentTextCharacter = String(currentText[currentTextRange])
+                    let currentPatternCharacter = getCharacterAt(index.patternIndex, pattern)
```

Comment:
use `let currentPatternCharacter = pattern[index.patternIndex]` instead of

Reply:
not supported by swift